refactor(allocator): re-order code

overlookmotel opened the pull request:
Follow-on after #11736. Pure refactor. Move all code related to `AllocatorGuard` to be together, so it's easier to read. Also rearrange `std` imports.

camc314 commented:
### Merge activity

* **Jun 16, 9:23 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(allocator): re-order code (#11759)

Follow-on after #11736. Pure refactor. Move all code related to `AllocatorGuard` to be together, so it's easier to read. Also rearrange `std` imports.
